docker-app/qfieldcloud/core/cron.py

```diff
@@ -35,3 +43,33 @@ def do(self):
         logger.info(
             f'Resend {len(invitation_emails)} previously failed invitation(s) to: {", ".join(invitation_emails)}'
         )
+
+
+class SetTerminatedWorkersToFinalStatusJob(CronJobBase):
+    # arbitrary number 3 here, it just feel a good number since the configuration is 10 mins
+    schedule = Schedule(run_every_mins=3)
+    code = "qfieldcloud.set_terminated_workers_to_final_status"
+
+    def do(self):
+
+        client = docker.from_env()
+        qgis_containers = client.containers.list(
+            sparse=True, filters={"ancestor": QGIS_CONTAINER_NAME}
+        )
+        qgis_container_ids = [c.id for c in qgis_containers]
+
+        jobs = Job.objects.filter(
+            status__in=[Job.Status.QUEUED, Job.Status.STARTED],
+        ).exclude(container_id__in=qgis_container_ids)
+
+        jobs.update(
+            status=Job.Status.FAILED,
+            finished_at=timezone.now(),
+            feedback={
+                "error_stack": "",
+                "error": "Worker unexpectedly terminated.",
+                "error_origin": "worker_wrapper",
+                "container_exit_code": -2,
+            },
+            output="Job unexpectedly terminated.",
+        )
```

Comment:
Does this output an error to Sentry ? If not, I'd recommend to do so, so we know something is wrong (or are tasks monitored in a different way)